### ISSUE TITLE
Basic Weapon

### ISSUE THREAD

**Issue by DrakeRyzer** [OmniBrew/sw5e-data#3]
Guessing images need to be uploaded somewhere specific to work. And then the Damage Type, Rarities, and weapon_types sections. I am guessing those have some base code that needs to be grabbed? I think part of it is because ME5e does weapons differently so I guess I will need to figure out how to separate that. Maybe the current table uses Blaster, Lightweapon, or Vibroweapon in place of the ME5e weapon type lines?

![image](https://github.com/OmniBrew/sw5e-data/assets/71940233/425b042f-9673-4111-b418-8f8cb95f5b81)

![image](https://github.com/OmniBrew/sw5e-data/assets/71940233/ae205501-45ad-44c1-9b18-60fc9ac17992)

**Comment by queryluke** [comment]
Hey Drake! I'm not sure what @bdc4 has explained so I apologize if any of this is a repeat.

### RE: Images
Getting images from Fandom can be a pain. Sometimes the static link works and sometimes it doesn't. Once you've got the image in a browser tab (or copied), try changing the `static` in `https://static.wikia.nocookie...` to `vignette`. So, with the blaster pistol example: `https://vignette.wikia.nocookie.net/starwars/images/8/85/DL18BlasterPistol-SWBF.png/revision/latest?cb=20160518002943`. I have no clue what the differences are, but I've found the vignette links to be more stable.

### RE: Damage types & Weapon Types
It looks like you figured this one out. I can see the correct damage and weapon types on the test site now.

### RE: Rarities
For rarities, you just need to add a new line for unenhanced in the rarities section of the messages file: https://github.com/OmniBrew/sw5e-data/blob/main/text/en/messages/index.js#L330

Also, a note about rarities. The way it's currently set up, if you want to have the "Artifact" word show up as a rarity, you'd have to put "spectre" in the rarity section. Or "very_rare" = "Legendary". Since that might get confusing, feel free to completely change that section to match the SW5e rarities, in both key and value.

```js
    rarities: {
      varies: "Varies", // not sure if you even need this
      unenhanced: "Unenhanced",
      standard: "Standard",
      premium: "Premium",
      prototype "Prototype",
      legendary: "Legendary",
      artifact "Artifact",
    },
```

We'll have to make a couple of changes on the frontend to get the colors working, but it's a simple change.

@bdc4 if you want to make the change on the frontend, it's this file: https://github.com/OmniBrew/sw5e/blob/master/store/config.js#L21

Note there are colors and text colors. I don't think the color section is used anymore (it used to be for icons). The text colors, you'll want to be sure that the ones in the "light" section work with a white background and the ones in the "dark" section work with a black background.

**Comment by DrakeRyzer** [comment]
Sounds good, I think Heat needs to be changed on the front end for the filters as well? I tried changing all instances of Heat to Reload that I could, but it just displayed incorrectly (or not at all) on the weapon table. I got it to say Reload just fine now, but it doesn't fill in that section still.
![image](https://github.com/OmniBrew/sw5e-data/assets/71940233/3c117462-a0c2-4e0b-98d1-a13c0fa9a480)
![image](https://github.com/OmniBrew/sw5e-data/assets/71940233/1a3d80a4-f34a-43ce-b408-0dbbad8e5d98)

**Comment by DrakeRyzer** [comment]
I also have the rarity colors we currently use (or as accurate as I could), but I am free on changing Standard if it doesn't look good
Black/White = Unenhanced/Basic/Varies
Gray = #4A5568 = Standard
Green = #007900 = Premium
Blue = #0000FD = Prototype
Purple = #800080 = Advanced
Orange = #DD6B20 = Legendary
Red = #FF0000 = Artifact

**Comment by queryluke** [comment]
Yeah, in the frontend, there are a lot of references to `weapon.heat` so we'll need to find those and replace with `reload`. But I updated the one place on the weapon display that shows it for now.

I also updated the rarity key/value pairs as well (as I mentioned in my previous comment). And add the colors to the site.

One note about weapon properties. For them to show up they need to have a file in the `/data/weapon-properties/` directory. You can just copy and rename existing weapon props then end the necessary data in the file (for example, I added power-cell).

Then, in the weapon file, you reference those files using the file name without the extension. So `power-cell` would reference the `power-cell.md` file.

And the formatting for lists/arrays is either:

```yaml
properties: [power-cell, two-handed] # note the square brackets
```

or 

```yaml
properties:
  - power-cell  # each dash - indicates a new item in the list
  - two-handed
```
